lib/FunctionIsolation/InvokeIsolatedFunctions.cpp

```diff
@@ -190,15 +163,21 @@ class InvokeIsolatedFunctions {
         for (const auto &ArgumentLayout : Layout.Arguments) {
           for (model::Register::Values Register : ArgumentLayout.Registers) {
             auto Name = model::Register::getCSVName(Register);
-            GlobalVariable *CSV = M->getGlobalVariable(Name, true);
+            GlobalVariable *CSV = RootModule.getGlobalVariable(Name, true);
             revng_assert(CSV != nullptr);
             Arguments.push_back(createLoad(Builder, CSV));
           }
         }
       }
 
+      llvm::Function *
+        NewDeclaration = llvm::Function::Create(F->getFunctionType(),
```

Comment:
Using function types of another module scares me, but it should be fine :thinking:

Reply:
I looked around and it looks like this is safe to do as long as both modules share the same context. Which we do here, I assume. Not to mention the two modules will be merged together right after this!